File: Minimal/PHP/url_shortener.php
<?php

class URLShortener {
    private $urls = [];

    public function shorten($url) {
        // Generate random 6-character short code
        $short = $this->generateShortCode(6);
        $this->urls[$short] = $url;
        return $short;
    }

    public function getUrl($short) {
        return isset($this->urls[$short]) ? $this->urls[$short] : null;
    }

    private function generateShortCode($length) {
        $characters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
        $short = '';
        for ($i = 0; $i < $length; $i++) {
            $short .= $characters[rand(0, strlen($characters) - 1)];
        }
        return $short;
    }
}

// Example usage
if (basename(__FILE__) == basename($_SERVER["SCRIPT_NAME"])) {
    $shortener = new URLShortener();
    $short = $shortener->shorten("https://www.example.com");
    echo "Short URL: " . $short . "\n";
    $original = $shortener->getUrl($short);
    echo "Original URL: " . $original . "\n";
}

?>

File: Minimal/Python/url_shortener.py
import hashlib
import random
import string

class URLShortener:
    def __init__(self):
        self.urls = {}

    def shorten(self, url):
        # Generate random short code (6 characters)
        short = ''.join(random.choices(string.ascii_letters + string.digits, k=6))
        self.urls[short] = url
        return short

    def get_url(self, short):
        return self.urls.get(short)

# Example usage
if __name__ == "__main__":
    shortener = URLShortener()
    short = shortener.shorten("https://www.example.com")
    print(f"Short URL: {short}")
    original = shortener.get_url(short)
    print(f"Original URL: {original}")